packages/ts/src/core/container/index.ts

```diff
@@ -42,6 +44,13 @@ export class ContainerCore {
       .attr('height', ContainerCore.DEFAULT_CONTAINER_HEIGHT) // Overriding default SVG height of 150
       .attr('aria-hidden', true)
 
+    this._svgDefs = this.svg.append('defs')
+
+    if (select(`defs.${defsLabel}`).empty()) {
```

Comment:
I find this line with the global `select` a bit confusing. I guess it's to prevent duplication of defs when there's more than 1 chart on the page. If so, it would benefit from having a comment with description.

Also it can be error prone: if a component containing the defs is removed from the page (e.g. user collapsed some view), other charts won't be able to find the defs anymore. Maybe it's worth moving this logic to `render`